antlr-kotlin-runtime-common/src/main/kotlin/com/strumenta/kotlinmultiplatform/Misc.kt

```diff
@@ -47,10 +47,9 @@ expect class BitSet {
 
 expect object Collections {
     fun unmodifiableList(asList: Collection<*>): List<*>
-    fun <T, U> unmodifiableMap(t: T): U
+    fun <T, U> unmodifiableMap(map: Map<T, U>): MutableMap<T, U>
```

Comment:
The method is called unmodifiable map and then it returns a MutableMap. It is surprising. We could probably rename it or change the type returned

Reply:
Changing the return type leads to errors in Recognizer.kt:
e: antlr-kotlin/antlr-kotlin-runtime-common/src/main/kotlin/org/antlr/v4/kotlinruntime/Recognizer.kt: (93, 42): Type inference failed. Expected type mismatch: inferred type is Map<String, Int> but MutableMap<String, Int>? was expected
e: antlr-kotlin/antlr-kotlin-runtime-common/src/main/kotlin/org/antlr/v4/kotlinruntime/Recognizer.kt: (114, 42): Type inference failed. Expected type mismatch: inferred type is Map<String, Int> but MutableMap<String, Int>? was expected